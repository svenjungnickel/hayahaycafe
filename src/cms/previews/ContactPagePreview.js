import React from 'react';
import PropTypes from 'prop-types';
import { ContactPageTemplate } from '../../templates/ContactPage';

const ContactPagePreview = ({ entry, widgetFor }) => (
    <ContactPageTemplate
        title={entry.toJS().data.title}
        subtitle={entry.toJS().data.subtitle}
        headerImage={entry.toJS().data.headerImage}
        content={widgetFor('body')}
        what3WordsAddress={entry.toJS().data.what3WordsAddress}
        what3wordsIcon={entry.toJS().data.what3wordsIcon}
    />
);

ContactPagePreview.propTypes = {
    entry: PropTypes.shape({
        getIn: PropTypes.func,
    }).isRequired,
    widgetFor: PropTypes.func.isRequired,
};

export default ContactPagePreview;
