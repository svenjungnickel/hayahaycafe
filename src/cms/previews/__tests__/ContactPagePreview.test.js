import React from 'react';
import renderer from 'react-test-renderer';
import ContactPagePreview from '../ContactPagePreview';
import ContactPageData from '../../../__fixtures__/ContactPageData';

jest.mock('../../../components/Header');
jest.mock('../../../components/ContactForm/ContactForm');
jest.mock('../../../components/Location/Location');
jest.mock('../../../components/Separator');
jest.mock('../../../components/What3WordsAddress');
jest.mock('../../../components/OpeningHours');

describe('ContactPagePreview', () => {
    it('renders without entry throws prop type error', () => {
        const renderContactPagePreview = () => {
            const widgetFor = () => {};

            const component = <ContactPagePreview widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderContactPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders without widgetFor throws prop type error', () => {
        const renderContactPagePreview = () => {
            const entry = {
                toJS: () => {},
            };

            const component = <ContactPagePreview entry={entry} />;
            renderer.create(component);
        };

        expect(renderContactPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid entry throws prop type error', () => {
        const renderContactPagePreview = () => {
            const entry = {
                toJS: 123,
            };
            const widgetFor = () => {};

            const component = <ContactPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderContactPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid widgetFor throws prop type error', () => {
        const renderContactPagePreview = () => {
            const entry = {
                toJS: () => {},
            };
            const widgetFor = {};

            const component = <ContactPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderContactPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const entry = {
            toJS: () => {
                return {
                    data: ContactPageData,
                };
            },
        };
        const widgetFor = () => 'content';

        const component = <ContactPagePreview entry={entry} widgetFor={widgetFor} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
