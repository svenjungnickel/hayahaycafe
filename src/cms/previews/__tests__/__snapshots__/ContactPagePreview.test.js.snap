// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPagePreview renders correctly 1`] = `
Array [
  <header />,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="col-sm-4 col-12"
        >
          <p>
            Location
          </p>
          <hr />
          <a
            href="what3WordsAddress"
          >
            <img
              alt="what3WordsIcon"
              src="src"
            />
          </a>
          <hr />
          <p>
            Opening Hours
          </p>
          <hr />
        </div>
        <div
          className="col-sm-8 col-12"
        >
          <h2
            className="d-block d-sm-none"
          >
            Contact us
          </h2>
          <form />
        </div>
      </div>
    </div>
  </section>,
]
`;
